### PR TITLE
Dev 519 resumption tokens after first page

### PR DESCRIPTION
**Pull request by billdueber**
Added a subclass of the stock ResumptionToken
and use new methods to easily create a new
token from an old one. Code in model and
partial_result now call the same method instead
of duplicating code, and allows us to put local-specific
stuff (e.g., that a new token has a last_str of "*")
in one spot.

Also extracted default query params and default
token params, and added tests to verify that we're getting
next tokens and different results for successive pages.

### CODE REVIEW

File: lib/oai_solr/partial_result.rb [hunk 1]
```diff
@@ -28,11 +28,19 @@ def records
...
 
+    # Build a new OAI::Provider::ResumptionToken based on information from the existing
+    # token (if any) and return it.
     # @return [OAI::Provider::ResumptionToken] The resumption token object with data pulled from
     #   @opts and @response
     def token
+      old_rt_opts = if @opts[:resumption_token]
+        OAI::Provider::ResumptionToken.parse(@opts[:resumption_token]).to_conditions_hash
+      else
+        {prefix: "oai_dc"}
```
[the remark this answers is not in the document]
Comment (aelkiss) [reply lib/oai_solr/partial_result.rb:39]: Also isn't this already what `https://github.com/hathitrust/oai_solr/blob/4ec4b55bb1ce35aa596489ee2ad35c59180a1498/lib/oai_solr/model.rb#L53` is (clunkily) doing? I think there is some opportunity to clean up `OAISolr::Model` and reduce the duplication..

[the remark this answers is not in the document]
Comment (billdueber) [reply lib/oai_solr/partial_result.rb:39]: You're right, of course, on both counts. I'll figure out a decent way abstract it out.

**Comment by aelkiss** [comment]
Looks pretty good to me. I will deploy in testing and make sure it's working as expected.

File: lib/oai_solr/partial_result.rb [hunk 2]
```diff
@@ -28,14 +29,14 @@ def records
...
+    # @return [OAISolr::ResumptionToken] The resumption token object with data pulled from
     #   @opts and @response
     def token
-      OAI::Provider::ResumptionToken.new(
-        @opts.merge(last: @response["nextCursorMark"]),
-        nil,
-        @response["response"]["numFound"]
-      )
+      total = @response["response"]["numFound"]
+      opts = @opts.merge(last: @response["nextCursorMark"])
```
Comment (aelkiss) [inline lib/oai_solr/partial_result.rb:38]: Something doesn't quite seem to be working as expected here. If I visit (for example) https://testing.oai.kubernetes.hathitrust.org/oai?verb=ListRecords&resumptionToken=marc21.f(1970-01-01T00:00:00Z).u(2022-12-07T18:39:56Z):AoEpMDAwMDAwNjA5 

then I get exactly the same `resumptionToken` back and never progress beyond that page.

**Review by aelkiss — CHANGES_REQUESTED** [review]
`nextCursorMark` doesn't seem to be updating; not sure if that's because it isn't making it to solr or the next cursor mark isn't being put correctly in the output token. Should be easy enough to test that we get a different resumption token back than the one we fed in.

File: lib/oai_solr/partial_result.rb (hunk 2 above)
Comment (billdueber) [reply lib/oai_solr/partial_result.rb:38]: I'll track it down. I was working too late to get the details right, apparently. Test will be easy to add.

**Review by aelkiss — APPROVED** [review]
Looking good to me now on testing.oai.hathitrust.org (first 20 pages of results, anyway).

Please go ahead and merge when ready.

**Pull request merged**